plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -327,6 +351,62 @@ func (s *SQLServer) accRow(query Query, acc telegraf.Accumulator, row scanner) e
 	return nil
 }
 
+// gatherHealth stores info about any query errors in the healthMetrics map
+func (s *SQLServer) gatherHealth(healthMetrics map[string]*HealthMetric, serv string, queryError error, mutex *sync.Mutex) {
+	if !s.HealthMetric {
+		return
+	}
+
+	mutex.Lock()
```

Comment:
I understand why we're trying to do this. Does this mean that now we're serializing instead of parallel? Even though gatherServer is parallel, all processes will need to execute gatherHealth in a serial fashion right coz of the lock? If yes, do you think having pub-sub mechanism will help to retain the parallelism? Like gatherServer publises the queryError and notifies gatherHealth whenever there are new records?

Reply:
Queries are where a bulk of the time are spent. Queries are still executed in parallel. Once queries are complete, their results are added to the dictionary in serial. That critical section is pretty small. I don't see it causing any more than ones of milliseconds of delay. 

I'd be in favor of keeping this as-is (with a single lock and dictionary). Other strategies would add more complexity than would be needed for this kind of simple parallelization.

Reply:
ok, as discussed maybe we can add some testing results to know the time spent in this block. (timer to keep track of time during its execution in this mutex). Negligible amount of delay as you called out ( almost 0 or ones of milliseconds) might be ok.